src_Core/CPU/CPU_Fetch_C.bsv

```diff
@@ -222,8 +239,24 @@ module mkCPU_Fetch_C #(IMem_IFC  imem32) (IMem_IFC);
 	  && (addr_of_b32 == imem32.pc)
 	  && is_32b_instr (imem32.instr [31:16]))
 	 begin
-	    addr_of_b32 = addr_of_b32 + 4;
+	    addr_of_b32 = imem32.pc + 4;
+            // Here, we know that addr[0:1] == 2'b10 (from is_addr_odd16).
+            // We also know that addr_of_b32 = addr with the bottom 2 bits set to 2'b0
+            // ie addr_of_b32 = addr - 2       (1)
+            // Since after this if statement we make a request to imem32.req with the
+            // address addr_of_b32, the next time imem32.valid is True we will have
+            // imem32.pc = addr_of_b32 + 4     (2)
+            // so we will have imem32.pc = addr + 2     using (1) and (2)
+            // since we set rg_pc = addr above, we then get
+            // imem32.pc = rg_pc + 2
+            rg_imem_pc_is_rg_pc_plus_2 <= True;
 	 end
+         else begin
+            // This is similar to the first branch of this if statement, but we get either
+            // imem32.pc = rg_pc  or  imem32.pc = rg_pc - 2
+            // In either case we can assert this to be False
+            rg_imem_pc_is_rg_pc_plus_2 <= False;
+         end
```

Comment:
These additions all need the first 8 spaces turning into a tab.

Reply:
Thanks for the pointer, and sorry it took so long to fix this. It should be fixed now.